docs/LoaderLayerInterface.md

```diff
@@ -412,14 +419,136 @@ following:
   /usr/share/vulkan/implicit_layer.d
 ```
 
+### Layer Filtering
+
+There are new environment variables that will allow more control over what
+layers the loader may load.
+These should also make it easier to diagnose layer problems or restrict usage
+of any layers that may cause problems.
+
+The new layer enable environment variable `VK_LOADER_LAYERS_ENABLE` is a
+comma-delimited list of globs to search for in known layers.
+Known layers are those that are already found by the loader taking into account
+default search paths and other environment variables (like
+`VK_LAYER_PATH` or `VK_ADD_LAYER_PATH`).
+The layer names are compared against the globs listed in the environment
+variable, and if they match, they will automatically be added to the enabled
+layer list in the loader for each application.
+These layers are enabled after implicit layers but before other explicit layers.
+
+When a layer is enabled using the new `VK_LOADER_LAYERS_ENABLE` filter, and
+loader logging is set to emit either warnings or layer messages, then a message
+will show for each layer that has been forced on.
+This message will look like the following:
+
+```
+WARNING | LAYER:  Layer "VK_LAYER_LUNARG_wrap_objects" force enabled due to env var 'VK_LOADER_LAYERS_ENABLE'
+```
+
+The new layer disable environment variable `VK_LOADER_LAYERS_DISABLE` is a
+comma-delimited list of globs to search for in known layers.
+Known layers are those that are already found by the loader taking into account
+default search paths and other environment variables
+(like `VK_LAYER_PATH` or `VK_ADD_LAYER_PATH`).
+The layer names are compared against the globs listed in the environment
+variable, and if they match, they will automatically be disabled (whether or not
+the layer is Implicit or Explicit).
+This means that they will not be added to the enabled layer list in the loader
+for each application.
+This could mean that layers requested by an application are also not enabled
+such as `VK_KHRONOS_LAYER_synchronization2` which could cause some applications
+to misbehave.
```

Comment:
If an application asks for layer X to be enabled in instance creation, but VK_LOADER_LAYERS_DISABLE is set to disable that layer, wouldn't the app get VK_ERROR_LAYER_NOT_PRESENT in instance creation? That would be my understanding. 

Put another way, does vkEnumerateInstanceLayerProperties include disabled layers? If not, then the app shouldn't 'try' to enable a layer that doesn't exist according to vkEnumerateInstanceLayerProperties.

Reply:
So vkEnumerateInstanceLayerProperties no longer shows that layer.  So, if they are looking for the layer, it won't be there.  If they blindly enable it, vkCreateInstance ignores that layer inside the driver and only reports messages via it's log.

Reply:
Doing a bit of playing around, it seems that `VK_LOADER_LAYERS_DISABLE` disabling a layer that the application is trying to enable in `VkInstanceCreateInfo` does *not* provoke `vkCreateInstance` returning an error. It disables the layer and lets the app create an instance.

However, when calling `vkEnumerateInstanceLayerProperties()` the layer is still present in the list. I think this will cause confusion because the layers apps try to enable will appear when they look for it but aren't enableable.

Reply:
The behavior that was settled on was to fail instance creation if an application tries to enable a 'disabled' layer with VK_ERROR_LAYER_NOT_PRESENT, as well as 'filter' out disabled layers from `vkEnumerateInstanceLayerProperties()`.